sdk/inc/azure/core/internal/az_http_internal.h

```diff
@@ -217,8 +242,11 @@ AZ_NODISCARD az_result az_http_request_append_path(az_http_request* ref_request,
  *     - `name` or `value` are empty.
  *     - `name`'s or `value`'s buffer overlap resulting `url`'s buffer.
  */
-AZ_NODISCARD az_result
-az_http_request_set_query_parameter(az_http_request* ref_request, az_span name, az_span value);
+AZ_NODISCARD az_result az_http_request_set_query_parameter(
+    az_http_request* ref_request,
+    az_span name,
+    az_span value,
+    bool is_query_url_encoded);
```

Comment:
Should we consider creating a URL struct type that has a url span and is_encoded bool field?

Reply:
> The reason why we separate bools is because it is likely that the name comes from flash memory and is already url encoded but the value comes from arbitrary input and encoding is required.

In that case, maybe we don't have the `url_encode_name` parameter, and only have `url_encode_value`, which applies just to the value.

This matches .NET's `public void AppendQuery(string name, string value, bool escapeValue)`, which only allows escaping the value, and not the name.